src/DBFTPlugin/ConsensusContext.cs

```diff
@@ -335,31 +337,17 @@ internal int GetExpectedBlockSizeWithoutTransactions(int expectedTransactions)
         /// <param name="txs">Ordered transactions</param>
         internal void EnsureMaxBlockLimitation(IEnumerable<Transaction> txs)
         {
-            uint maxBlockSize = NativeContract.Policy.GetMaxBlockSize(Snapshot);
-            long maxBlockSystemFee = NativeContract.Policy.GetMaxBlockSystemFee(Snapshot);
-            uint maxTransactionsPerBlock = NativeContract.Policy.GetMaxTransactionsPerBlock(Snapshot);
+            uint maxTransactionsPerBlock = system.Settings.MaxTransactionsPerBlock;
 
             // Limit Speaker proposal to the limit `MaxTransactionsPerBlock` or all available transactions of the mempool
             txs = txs.Take((int)maxTransactionsPerBlock);
             List<UInt256> hashes = new List<UInt256>();
             Transactions = new Dictionary<UInt256, Transaction>();
             VerificationContext = new TransactionVerificationContext();
 
-            // Expected block size
-            var blockSize = GetExpectedBlockSizeWithoutTransactions(txs.Count());
-            var blockSystemFee = 0L;
-
             // Iterate transaction until reach the size or maximum system fee
             foreach (Transaction tx in txs)
             {
-                // Check if maximum block size has been already exceeded with the current selected set
-                blockSize += tx.Size;
-                if (blockSize > maxBlockSize) break;
-
-                // Check if maximum block system fee has been already exceeded with the current selected set
-                blockSystemFee += tx.SystemFee;
-                if (blockSystemFee > maxBlockSystemFee) break;
```

Comment:
I think the `MaxBlockSize` and `MaxBlockSysFee` should be moved to the plugin's settings.

Reply:
Fixed.